### ISSUE TITLE
[BUG] Exception in thread "inform_stat-1" java.lang.StackOverflowError

### ISSUE THREAD

**Issue by JohnLahr** [linuxserver/docker-unifi-controller#182]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After some time (could be minutes or hours), the UniFi controller starts to throw hundreds of thousands of lines of stacktraces that very quickly fill up the container's storage. I don't notice any performance issues or other failures within the controller itself, so it's hard to pinpoint what is triggering these errors.

### Expected Behavior

This shouldn't happen.

### Steps To Reproduce

1. Start container
2. Wait

### Environment

```markdown
- OS: Ubuntu Server 22.04
- How Docker service was installed: Ansible playbook (sorry...)


---
# --------------------------------------------------------------------------- #
# UniFi Controller Tasks                                                      #
# --------------------------------------------------------------------------- #

- name: Ensure UniFi Controller config directory exists
  ansible.builtin.file:
    path  : '{{ container_storage_root }}/unifi/config'
    state : directory
    owner : '{{ unifi_controller_puid }}'
    group : '{{ unifi_controller_pgid }}'
    mode  : '0700'

- name: Define UniFi controller container
  community.docker.docker_container:
    command_handling: correct
    name: unifi
    hostname: '{{ unifi_controller_hostname }}'
    domainname: '{{ domain_name }}'
    image: linuxserver/unifi-controller
    restart_policy: unless-stopped
    network_mode: host
    dns_search_domains:
      - '{{ domain_name }}'
    dns_servers: '{{ private_dns_servers }}'
    mounts:
      - type: bind
        source: '{{ container_storage_root }}/unifi/config'
        target: /config
    env:
      PUID      : '{{ unifi_controller_puid }}'
      PGID      : '{{ unifi_controller_pgid }}'
      MEM_LIMIT : '{{ unifi_controller_memory_limit }}'
      TZ        : '{{ unix_timezone }}'
    labels:
      'com.centurylinklabs.watchtower.enable': 'true'

# --------------------------------------------------------------------------- #
```
```


### CPU architecture

x86-64

### Docker creation

```bash
tl;dr - ansible-playbook -i "${inventoryFile}" "${playbookFile}" -b --vault-password-file "${VAULT_PASSWORD_FILE}"

full script:

#!/bin/bash
#=============================================================================#
# HEADER                                                                      #
#=============================================================================#
#% SYNOPSIS                                                                  %#
#+    ${SCRIPT_NAME} ANSIBLE_INVENTORY_FILE ANSIBLE_PLAYBOOK_FILE [args ...] +%
#%                                                                           %#
#% DESCRIPTION                                                               %#
#%    Ansible playbook execution script.                                     %#
#%                                                                           %#
#% OPTIONS                                                                   %#
#%    -h, --help                    Print this help                          %#
#%                                                                           %#
#=============================================================================#
#  DEBUG OPTION                                                               #
#    set -n  # Uncomment to check syntax without execution                    #
#    set -x  # Uncomment to debug shell script                                #
#                                                                             #
#=============================================================================#
# END_OF_HEADER                                                               #
#=============================================================================#

set -o pipefail
set -e

# --------------------------------------------------------------------------- #

readonly GPG_EMAIL="<REDACTED>"
readonly VAULT_PASSWORD_FILE="vault_password.txt"
readonly ENCRYPTED_VAULT_PASSWORD_FILE="${VAULT_PASSWORD_FILE}.enc"

declare -a DECRYPTED_FILES

# --------------------------------------------------------------------------- #

readonly COLOR_NONE='\033[0m'

readonly COLOR_RED='\033[0;31m'
readonly COLOR_GREEN='\033[0;32m'
readonly COLOR_ORANGE='\033[0;33m'

function logError() {
    echo -e "[${COLOR_RED}ERROR${COLOR_NONE}] $1"
}

function logInfo() {
    echo -e "[${COLOR_GREEN}INFO${COLOR_NONE}] $1"
}

function logWarning() {
    echo -e "[${COLOR_ORANGE}WARNING${COLOR_NONE}] $1"
}

# --------------------------------------------------------------------------- #
# Displays help menu
# --------------------------------------------------------------------------- #
function show-help() {
    cat << EOT
USAGE:
    $(basename $0) ANSIBLE_INVENTORY_FILE ANSIBLE_PLAYBOOK_FILE [args ...]

DESCRIPTION:
    Ansible playbook execution script.

OPTIONS
    -h, --help                    Print this help

EXAMPLES
    $(basename $0) inventory.yaml playbooks/unifi-controller.yaml
EOT
}

# --------------------------------------------------------------------------- #
# Decrypts a file using the default GPG key.
#
# @example
#   decrypt /path/to/file
#
# @arg $1 Path to encrypted file
# --------------------------------------------------------------------------- #
function decrypt() {
    local -r encryptedFile="$1"
    local -r decryptedFile="${1%.enc}"

    gpg --decrypt \
        --output "${decryptedFile}" \
        "${encryptedFile}"

    rm "${encryptedFile}"

    DECRYPTED_FILES+="${decryptedFile}"
}

# --------------------------------------------------------------------------- #
# Decrypts a file using the default GPG key.
#
# @example
#   decrypt /path/to/file
#
# @arg $1 Path to encrypted file
# --------------------------------------------------------------------------- #
function encrypt() {
    local -r encryptedFile="$1".enc
    local -r decryptedFile="$1"

    gpg --encrypt \
        --recipient "${GPG_EMAIL}" \
        --output "${encryptedFile}" \
        "${decryptedFile}"

    rm "${decryptedFile}"

    DECRYPTED_FILES=( "${DECRYPTED_FILES[@]/$decryptedFile}" )
}

# --------------------------------------------------------------------------- #
# Finds and decrypts encrypted files
# --------------------------------------------------------------------------- #
function decryptSecrets() {
    local encryptedFiles=()

    while IFS= read -d $'\0' -r file ; do
        encryptedFiles=("${encryptedFiles[@]}" "${file}")
    done < <(find . -name "*.enc" -print0)

    logInfo "Decrypting encrypted deployment files"

    for file in "${encryptedFiles[@]}"; do
        decrypt "${file}"
    done
}

# --------------------------------------------------------------------------- #
# Re-encrypts previously decrypted files
# --------------------------------------------------------------------------- #
function reencryptSecrets() {
    if [[ "${DECRYPTED_FILES[@]}" =~ ^\ +$ ]]; then
        logWarning "Attempted to reencrypt secrets, but no decrypted secrets were found"
    else
        logInfo "Reencrypting previously decrypted deployment files"

        for file in "${DECRYPTED_FILES[@]}"; do
            encrypt "${file}"
        done
    fi
}

# --------------------------------------------------------------------------- #
# Executes an Ansible playbook
#
# @example
#   executePlaybook inventory.yaml playbooks/example.yaml
#
# @arg $1 Path to Ansible inventory file
# @arg $2 Path to Ansible playbook
# --------------------------------------------------------------------------- #
function executePlaybook() {
    local -r inventoryFile="$1"
    local -r playbookFile="$2"

    if [[ -z "${inventoryFile}" || -z "${playbookFile}" ]]; then
        logError "Must specify both an inventory file and a playbook file."
        exit 2
    fi

    if ! ansible-playbook \
        -i "${inventoryFile}" \
        "${playbookFile}" \
        -b \
        --vault-password-file "${VAULT_PASSWORD_FILE}"
    then
        reencryptSecrets
        exit 1
    fi
}

# --------------------------------------------------------------------------- #

case "$1" in
    help|-h|--help|-help)
        show-help
        ;;
    *)
        decryptSecrets
        executePlaybook $*
        reencryptSecrets
        ;;
esac

exit 0

# --------------------------------------------------------------------------- #

trap "reencryptSecrets" INT

# --------------------------------------------------------------------------- #
```

### Container logs

```bash
[custom-init] No custom services found, skipping...
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service 00-legacy: starting
s6-rc: info: service 00-legacy successfully started
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/01-envfile
cont-init: info: /etc/cont-init.d/01-envfile exited 0
cont-init: info: running /etc/cont-init.d/01-migrations
[migrations] started
[migrations] no migrations found
cont-init: info: /etc/cont-init.d/01-migrations exited 0
cont-init: info: running /etc/cont-init.d/10-adduser
usermod: no changes

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/


Brought to you by linuxserver.io
-------------------------------------

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

cont-init: info: /etc/cont-init.d/10-adduser exited 0
cont-init: info: running /etc/cont-init.d/15-install
cont-init: info: /etc/cont-init.d/15-install exited 0
cont-init: info: running /etc/cont-init.d/20-config
*** Setting Java memory limit to 4096 ***
cont-init: info: /etc/cont-init.d/20-config exited 0
cont-init: info: running /etc/cont-init.d/30-keygen
cont-init: info: /etc/cont-init.d/30-keygen exited 0
cont-init: info: running /etc/cont-init.d/99-custom-files
[custom-init] No custom files found, skipping...
cont-init: info: /etc/cont-init.d/99-custom-files exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-mods: starting
s6-rc: info: service init-mods successfully started
s6-rc: info: service init-mods-package-install: starting
s6-rc: info: service init-mods-package-install successfully started
s6-rc: info: service init-mods-end: starting
s6-rc: info: service init-mods-end successfully started
s6-rc: info: service init-services: starting
s6-rc: info: service init-services successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun unifi (no readiness notification)
s6-rc: info: service legacy-services successfully started
s6-rc: info: service 99-ci-service-check: starting
[ls.io-init] done.
s6-rc: info: service 99-ci-service-check successfully started
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.catalina.loader.WebappClassLoaderBase (file:/usr/lib/unifi/lib/tomcat-embed-core-9.0.59.jar) to field java.io.ObjectStreamClass$Caches.localDescs
WARNING: Please consider reporting this to the maintainers of org.apache.catalina.loader.WebappClassLoaderBase
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Exception in thread "inform_stat-1" java.lang.StackOverflowError
	at com.ubnt.service.topology.S.öo0000(Unknown Source)
	at com.ubnt.service.topology.S.ÖO0000(Unknown Source)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.ubnt.service.topology.S.öo0000(Unknown Source)
	at com.ubnt.service.topology.S.ÖO0000(Unknown Source)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.ubnt.service.topology.S.öo0000(Unknown Source)
	at com.ubnt.service.topology.S.ÖO0000(Unknown Source)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)

(this block just repeats, although sometimes it switches things up)

	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.ubnt.service.topology.S.öo0000(Unknown Source)
	at com.ubnt.service.topology.S.ÖO0000(Unknown Source)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
Exception in thread "inform_stat-2" java.lang.StackOverflowError
	at com.ubnt.service.topology.S.ÖO0000(Unknown Source)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.ubnt.service.topology.S.öo0000(Unknown Source)
	at com.ubnt.service.topology.S.ÖO0000(Unknown Source)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.ubnt.service.topology.S.öo0000(Unknown Source)
	at com.ubnt.service.topology.S.ÖO0000(Unknown Source)
```

**Comment by Jeff777K** [comment]
I have the same issue on a Synology DS220+ and DS218+

Log in portainer :

[custom-init] No custom services found, skipping...
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service 00-legacy: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service 00-legacy successfully started
cont-init: info: running /etc/cont-init.d/01-envfile
cont-init: info: /etc/cont-init.d/01-envfile exited 0
cont-init: info: running /etc/cont-init.d/01-migrations
cont-init: info: /etc/cont-init.d/01-migrations exited 0
cont-init: info: running /etc/cont-init.d/10-adduser
[migrations] started
[migrations] no migrations found
-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/
Brought to you by linuxserver.io
-------------------------------------
To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------
User uid:    1033
User gid:    100
-------------------------------------
cont-init: info: /etc/cont-init.d/10-adduser exited 0
cont-init: info: running /etc/cont-init.d/15-install
*** installing unifi packages ***
Selecting previously unselected package unifi.
(Reading database ... 9373 files and directories currently installed.)
Preparing to unpack /app/unifi.deb ...
Unpacking unifi (7.3.76-19582-1) ...
Setting up unifi (7.3.76-19582-1) ...
invoke-rc.d: could not determine current runlevel
invoke-rc.d: policy-rc.d denied execution of start.
cont-init: info: /etc/cont-init.d/15-install exited 0
cont-init: info: running /etc/cont-init.d/20-config
cont-init: info: /etc/cont-init.d/20-config exited 0
cont-init: info: running /etc/cont-init.d/30-keygen
cont-init: info: /etc/cont-init.d/30-keygen exited 0
cont-init: info: running /etc/cont-init.d/99-custom-files
[custom-init] No custom files found, skipping...
cont-init: info: /etc/cont-init.d/99-custom-files exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-mods: starting
s6-rc: info: service init-mods successfully started
s6-rc: info: service init-mods-package-install: starting
s6-rc: info: service init-mods-package-install successfully started
s6-rc: info: service init-mods-end: starting
s6-rc: info: service init-mods-end successfully started
s6-rc: info: service init-services: starting
s6-rc: info: service init-services successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun unifi (no readiness notification)
s6-rc: info: service legacy-services successfully started
s6-rc: info: service 99-ci-service-check: starting
[ls.io-init] done.
s6-rc: info: service 99-ci-service-check successfully started
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.catalina.loader.WebappClassLoaderBase (file:/usr/lib/unifi/lib/tomcat-embed-core-9.0.59.jar) to field java.io.ObjectStreamClass$Caches.localDescs
WARNING: Please consider reporting this to the maintainers of org.apache.catalina.loader.WebappClassLoaderBase
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release


My docker-compose:

version: "2.1"
services:
  unifi-controller:
    image: ghcr.io/linuxserver/unifi-controller:latest
    container_name: unifi-controller

    environment:
      - PUID=1033
      - PGID=100
 #     - MEM_LIMIT=1024M #optional
    volumes:
      - /volume1/docker/unifi-controller:/config
      - /etc/localtime:/etc/localtime:ro

    ports:
      - 3478:3478/udp
      - 10001:10001/udp
      - 8080:8080
      - 8443:8443
 #     - 1910:1910/udp #optional
 #     - 8843:8843 #optional
 #     - 8880:8880 #optional
 #     - 6789:6789 #optional
 #     - 5514:5514/udp #optional
    networks:
         unifi-controller:
            ipv4_address: 172.20.0.2
    labels:
      - "com.centurylinklabs.watchtower.enable=true"
    restart: unless-stopped
    
networks:
    unifi-controller:
        external: true

**Comment by decibelava** [comment]
I stumbled across this while trying to find more information. I am having the exact issue although my system is running on a Google Compute instance and not Docker. Debain 11. UniFi version 7.3.76. Discovered the issue when I hit 22 GB of log files and my server locked up. No idea what is causing the issue or if this information helps or not.

**Comment by Toniob** [comment]
Same issue here, on a Debian 11, Unifi version 7.3.83. I just tried to export my configuration, uninstall unifi and remove its data, install it back and restore the backup. And topology still shows nothing and those errors are eating all the disk space available.

**Comment by thespad** [comment]
As far as I can tell, this isn't a container issue, it's an upstream bug with the Unifi controller and would need to be reported to them to have any hope of a resolution (I appreciate they're not the best at handling bug reports).
